returnn/datasets/multi_proc.py

```diff
@@ -254,23 +287,33 @@ def _get(seq_idx: int) -> Optional[DatasetSeq]:
                 elif msg == "init_seq_order":
                     if dataset is None:
                         dataset = init_dataset(dataset_dict)
-                    if worker_index == 0:
-                        # We are responsible to get the seq order and distrib it to all the other workers.
-                        assert other_worker_conns is not None
-                        dataset.init_seq_order(**kwargs)
-                        seq_order = dataset.get_current_seq_order()
-                        for i, worker_conn in enumerate(other_worker_conns):
-                            worker_conn.send(("seq_order_shard", seq_order[i + 1 :: len(other_worker_conns) + 1]))
-                        parent_conn.send(("num_seqs", len(seq_order)))
-                        # Now reset seq order for ourself (as the role of a normal worker).
-                        kwargs["seq_order"] = seq_order[0 :: len(other_worker_conns) + 1]
-                        kwargs.pop("seq_list", None)
+                    if sharding_method == "dedicated":
                         dataset.init_seq_order(**kwargs)
+                        try:
+                            num_seqs = dataset.num_seqs
+                        except NotImplementedError:
+                            num_seqs = None
+                        parent_conn.send(("num_seqs", num_seqs))
+                    elif sharding_method == "seq_order":
+                        if worker_index == 0:
+                            # We are responsible to get the seq order and distrib it to all the other workers.
+                            assert other_worker_conns is not None
+                            dataset.init_seq_order(**kwargs)
+                            seq_order = dataset.get_current_seq_order()
```

Comment:
WDYT: should we add a log line here if this fails w/ `NotImplementedError` to direct the user towards the other sharding option, which may be better supported?

Reply:
You can maybe wrap this in a `try`/`except Exception as exc` and throw some new `Exception(f"{self}: sub dataset {dataset} get_current_seq_order failed. Maybe try sharding_method='dedicated'") from exc` or so.